### DISCUSSION TITLE
Is there functionality for the Nth Weekday of a Month, i.e. 1st Monday of a month

### DISCUSSION THREAD

**Discussion by ShepTeck** [Hexagon/croner#190] [Q&A] [answered]
Am I missing something?  Is there a feature for # that allows for the first 'n' weekday of a month?

First off, this is great!  I am modifying it to work without npm as I need a complete stand-alone single file implementation for google apps script.

I have a Quartz Cron schedule of `0 0 5 ? 1/1 2#1 *` which is a legitimate code that is `At 05:00:00am, on the 1st Monday of the month, every month starting in January`; but I don't see any logic or substitutions for '#' in croner.

here is my version in gist with my edits:

https://gist.github.com/ShepTeck/bb6bda4f4a1692b5cc4a49715a41c776

-ps I added logic because 0 0/30 * 1/1 * ? * (At second :00, every 30 minutes starting at minute :00, every hour, every day starting on the 1st, every month) was causing too many requests and the browser to crash.

**Comment by Hexagon** [comment, marked as answer]
Hello! There is no nth-weekday functionality right now, it's just basic vixie cron with a few additions.

But you can actually run a script at nth weekday by using configuration `legacyMode: false` and specifying day of month 1-7, then select a weekday. 2nd week is 8-14 etc.

If you are interested in adding "real" nth weekday, a pull request is welcome 👍 

P.s. `?` means "the value of the first run in croner. You should just use * for wildcards.

**Reply by ShepTeck** [reply]
thank you very much, I'm ingesting a bunch (over 1,000) quartz cron codes and have to produce the associated array of n-dates (25+) so they can be given to a calendar app; so in the case of using legacyMode: false I would have to run more substitutions, so I may try to make my own logic.  If not I'll def use your suggestion.

**Comment by Hexagon** [comment]
This feature is pre-released, and tracked at issue [#198](https://github.com/Hexagon/croner/issues/198)
